src/PDFPasswordForm.tsx

```diff
@@ -0,0 +1,164 @@
+import React, {useState, useRef, useMemo, type ChangeEvent, type FormEventHandler} from 'react';
+import PropTypes from 'prop-types';
+
+import {pdfPasswordFormStyles as styles} from './styles';
+
+type Props = {
+    isPasswordInvalid: boolean;
+    isFocused: boolean;
+    onSubmit: (password: string) => void;
+    onPasswordUpdated: (password: string) => void;
+    onPasswordFieldFocused: (isFocused: boolean) => void;
+};
+
+const propTypes = {
+    /** If the submitted password is invalid (show an error message) */
+    isPasswordInvalid: PropTypes.bool,
+
+    /** Should focus to the password input  */
+    isFocused: PropTypes.bool.isRequired,
+
+    /** Notify parent that the password form has been submitted */
+    onSubmit: PropTypes.func,
+
+    /** Notify parent that the password has been updated/edited */
+    onPasswordUpdated: PropTypes.func,
+
+    /** Notify parent that a text field has been focused or blurred */
+    onPasswordFieldFocused: PropTypes.func,
```

Comment:
```suggestion
    onPasswordFieldFocus: PropTypes.func,
```

We usually write in "present tense" for callbacks.